user_external/lib/smb.php

```diff
@@ -43,7 +43,7 @@ public function checkPassword($uid, $password) {
 		$uidEscaped=escapeshellarg($uid);
 		$password=escapeshellarg($password);
 		$result=array();
-		$command=self::SMBCLIENT.' //'.$this->host.'/dummy -U'.$uidEscaped.'%'.$password;
+		$command=self::SMBCLIENT.' //'.escapeshellarg($this->host).' -U'.$uidEscaped.'%'.$password;
```

Comment:
Any reason to remove the "/dummy" part of the host? I guess it's a random share name, and I don't know if it will work properly without it.

In addition, I'd rather escape the whole argument (including the double slash).